src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/aarch64/AArch64ABI.java

```diff
@@ -72,5 +75,41 @@ public String name() {
         return SystemABI.ABI_AARCH64;
     }
 
-
+    @Override
+    public Optional<MemoryLayout> layoutFor(NativeType type) {
+        switch (type) {
```

Comment:
Could use a switch expression here. Same for the other impls of SystemABI::layoutFor